core/src/main/java/org/elasticsearch/search/aggregations/bucket/filters/InternalFilters.java

```diff
@@ -113,13 +114,34 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.endObject();
             return builder;
         }
+
+        @Override
+        public boolean equals(Object other) {
+            if (this == other) {
+                return true;
+            }
+            if (other == null || getClass() != other.getClass()) {
+                return false;
+            }
+            InternalBucket that = (InternalBucket) other;
+            return Objects.equals(key, that.key)
+                    && Objects.equals(keyed,  keyed)
+                    && Objects.equals(docCount, that.docCount)
+                    && Objects.equals(aggregations, that.aggregations);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(getClass(), key, keyed, docCount, aggregations);
+        }
     }
 
     private final List<InternalBucket> buckets;
     private final boolean keyed;
-    private Map<String, InternalBucket> bucketMap;
+    transient private Map<String, InternalBucket> bucketMap;
```

Comment:
I think that the build may complain about order of modifiers. Maybe add a comment on why this field is transient?